README.md

```diff
@@ -19,7 +19,134 @@ HeFQUIN is a query federation engine for heterogeneous federations of graph data
+```
+
+The optional build argument (`TAG`) can used to build the image from a specific release:
+```yml
+  hefquin:
+    build:
+      context: .
+      args:
+        TAG: <insert tag>
+    ports:
+      - "8080:8080"
+```
+
+The engine can be configured by mounting custom configuration files when starting up the container:
+```yml
+  hefquin:
+    build: .
+    ports:
+      - "8080:8080"
+    volumes:
+     - ./example/config.properties:/usr/local/tomcat/webapps/ROOT/config.properties
+     - ./example/ExampleEngineConf.ttl:/usr/local/tomcat/webapps/ROOT/ExampleEngineConf.ttl
+     - ./example/ExampleFedConf.ttl:/usr/local/tomcat/webapps/ROOT/ExampleFedConf.ttl
+```
+
+where the `config.properties` file has the following structure:
+```bash
+ENGINE_CONF_FILE=ExampleEngineConf.ttl   # the engine configuration file
+FED_CONF_FILE=ExampleFedConf.ttl         # the federation configuration file
+```
+
+By default the service exposes a SPARQL endpoint at `http://localhost:8080/sparql`.  
+</details>
+
+<details>
+  <summary><b>Run the server embedded in the JAR file</b></summary>
+
+__Option 1:__ Clone the project from the HeFQUIN repository and navigate to the project root in a terminal. Build the latest version of the engine using:
+```bash
+$ mvn clean package
+```
+To use a specific release, checkout the tag before building:
+```bash
+$ git checkout <insert tag>
+$ mvn clean package
+```
+
+__Option 2:__ Download a JAR file from the set of available releases (see [Releases](https://github.com/LiUSemWeb/HeFQUIN/releases)).
+
+Start the server in a terminal by running the following command:
+```bash
+java -cp target/HeFQUIN-x.y.z-SNAPSHOT.jar se.liu.ida.hefquin.service.HeFQUINServer
+```
+
+The engine can be configured by modifying the `config.properties` file in the working directory. The `config.properties` file has the following structure:
+```bash
+ENGINE_CONF_FILE=ExampleEngineConf.ttl   # the engine configuration file
+FED_CONF_FILE=ExampleFedConf.ttl         # the federation configuration file
+```
+</details>
+
+<details>
+  <summary><b>Run in a separate servlet container</b></summary>
+
+__Option 1:__ Clone the project from the HeFQUIN repository and navigate to the project root in a terminal. Build the latest version of the engine using:
+```bash
+$ mvn clean package
+```
+To use a specific release, checkout the tag before building:
+```bash
+$ git checkout <insert tag>
+$ mvn clean package
+```
+
+__Option 2:__ Download a JAR file from the set of available releases (see [Releases](https://github.com/LiUSemWeb/HeFQUIN/releases)).
+
+Deploy `target/HeFQUIN-x.y.z-SNAPSHOT.war` in your serlet container.
+
+The engine can be configured by modifying the `config.properties` file in the working directory. The `config.properties` file has the following structure:
+```bash
+ENGINE_CONF_FILE=ExampleEngineConf.ttl   # the engine configuration file
+FED_CONF_FILE=ExampleFedConf.ttl         # the federation configuration file
+```
```

Comment:
Like above, perhaps that's something to be moved to the detailed documentation pages a well.